tests/catch/include/hip_test_common.hh

```diff
@@ -23,39 +23,79 @@ THE SOFTWARE.
+    /*To see if error has occured in previous threads, stop execution to save time waiting for     \
+     * error*/                                                                                     \
+    if (TestContext::get().hasErrorOccured() == true) {                                            \
+      return; /*This will only work with std::thread and not with std::async*/                     \
+    }                                                                                              \
+    auto localError = error;                                                                       \
+    HCResult result(__LINE__, __FILE__, localError, #error);                                       \
+    TestContext::get().addResults(result);                                                         \
+  }
+
+#define REQUIRE_THREAD(condition)                                                                  \
+  {                                                                                                \
+    /*To see if error has occured in previous threads, stop execution to save time waiting for     \
+     * error*/                                                                                     \
+    if (TestContext::get().hasErrorOccured() == true) {                                            \
+      return; /*This will only work with std::thread and not with std::async*/                     \
     }                                                                                              \
+    auto localResult = (condition);                                                                \
+    HCResult result(__LINE__, __FILE__, hipSuccess, #condition, localResult);                      \
+    TestContext::get().addResults(result);                                                         \
   }
 
+// Do not call before all threads have joined
+#define HIP_CHECK_THREAD_FINALIZE()                                                                \
+  { TestContext::get().finalizeResults(); }
+
 #define HIPRTC_CHECK(error)                                                                        \
   {                                                                                                \
     auto localError = error;                                                                       \
-    if (localError != HIPRTC_SUCCESS) {                                                            \
-      INFO("Error: " << hiprtcGetErrorString(localError) << " Code: " << localError << " Str: "    \
-                     << #error << " In File: " << __FILE__ << " At line: " << __LINE__);           \
-      REQUIRE(false);                                                                              \
-    }                                                                                              \
+    INFO("Matching Error to HIPRTC_SUCCESS: "                                                      \
```

Comment:
I'm not sure we want to always display info if localError == success

Reply:
INFO is scoped and only printed when the condition that follows it (in the scope) is false.

So If REQUIRE, which follows INFO fails, the message will be printed.

Reply:
How is that even possible?

Reply:
This is a catch2 feature.

"Additional messages can be logged during a test case. Note that the messages logged with INFO are scoped and thus will not be reported if failure occurs in scope preceding the message declaration"

https://github.com/catchorg/Catch2/blob/devel/docs/logging.md